src/main/java/org/mozilla/zest/core/v1/ZestAssignFromElement.java

```diff
@@ -0,0 +1,253 @@
+    private int elementIndex;
+
+    private boolean returnElement;
+
+    private boolean returnAttribute;
+    private String returnedAttributeName;
+
+    public ZestAssignFromElement() {
+    }
+
+    public ZestAssignFromElement(String variableName) {
+        super(variableName);
+    }
+
+    @Override
+    public String assign(ZestResponse response, ZestRuntime runtime) throws ZestAssignFailException {
+        if (response == null) {
+            throw new ZestAssignFailException(this, "Null response");
+        }
+
+        Source src = new Source(response.getBody());
+        List<Element> elementsFilteredByElementName = filterByElementNameIfConfigured(src);
+        List<Element> elementsFilteredByAttributeValue = filterByAttributeValueIfConfigured(elementsFilteredByElementName);
+        List<Element> elements = reverseIfConfigured(elementsFilteredByAttributeValue);
+        Element element = findElementAtIndex(elements);
+        if(element == null){
+            return null;
+        }
+        return getReturnValue(element);
+    }
+
+    private String getReturnValue(Element element) throws ZestAssignFailException {
+        if(returnElement){
+            return element.getContent().toString();
+        }
+        else if(returnAttribute){
+            return element.getAttributeValue(returnedAttributeName);
+        }
+        else {
+            throw new ZestAssignFailException(this, "A selection method must be configured");
+        }
+    }
+
+    private Element findElementAtIndex(List<Element> elements) {
+        if(indexIsOutOfRange(elements)){
+            return null;
+        }
+        return elements.get(elementIndex);
+    }
+
+    private boolean indexIsOutOfRange(List<Element> elements) {
+        return elements.size() <= 0 || elementIndex >= elements.size();
+    }
+
+    private List<Element> reverseIfConfigured(List<Element> elements) {
+        if(reverseFilteredElements){
+            Collections.reverse(elements);
+            return elements;
+        }
+        return elements;
+    }
+
+    private List<Element> filterByElementNameIfConfigured(Source source){
+        if(filterElementName){
+            return source.getAllElements(elementNameFilter);
+        }
+        return source.getAllElements();
+    }
+
+    private List<Element> filterByAttributeValueIfConfigured(List<Element> elements) {
+        if (filterAttributeValue) {
+            List<Element> matchingElements = new ArrayList<>();
+            for (Element element: elements) {
+                if(isMatchingAttributeFilter(element)){
+                    matchingElements.add(element);
+                }
+            }
+            return matchingElements;
+        }
+        return elements;
+    }
+
+    private boolean isMatchingAttributeFilter(Element element) {
+        if(attributeValueFilterPattern == null){
+            attributeValueFilterPattern = Pattern.compile(attributeValueFilter, attributeValueFilterFlags);
```

Comment:
Should catch and wrap `IllegalArgumentException` in a `ZestAssignFailException`. Worth adding a test for this case.

Reply:
Actually the two cases (wrong flags and wrong regex).